src/plugins/requestAuthPlugin.js

```diff
@@ -26,20 +21,34 @@ function verifyAuth([request, auth]) {
 		throw new Error(errorMessage);
 	}
 	// there are secret tokens in `auth`, be careful with logging
-	request.log(['auth'], `Authorizing with keys: ${JSON.stringify(keys)}`);
+	request.log(['info', 'auth'], `Authorizing with keys: ${JSON.stringify(keys)}`);
 }
 
-function injectAuthIntoRequest([request, auth]) {
-	// update request with auth info
-	request.state.oauth_token = auth.access_token;  // this endpoint provides 'access_token' instead of 'oauth_token'
-	request.state.refresh_token = auth.refresh_token;  // use to get new oauth upon expiration
-	request.state.expires_in = auth.expires_in;  // TTL for oauth token (in seconds)
-
-	// special prop in `request.app` to indicate that this is a new,
-	// server-provided token, not from the original request, so the cookies
-	// will need to be set in the response
-	request.app.setCookies = true;
-}
+/**
+ * Both the incoming request and the outgoing response need to have an
+ * 'authorized' state in order for the app to render correctly with data from
+ * the API, so this function modifies the request and the reply
+ */
+const applyAuth = ([request, auth]) => {
+	const path = '/';
+	const authState = {
+		oauth_token: {
+			value: auth.oauth_token || auth.access_token,
+			opts: { path, ttl: auth.expires_in * 1000 },
+		},
+		refresh_token: {
+			value: auth.refresh_token,
+			opts: { path, ttl: YEAR_IN_MS * 2 },
+		}
+	};
+	Object.keys(authState).forEach(name => {
+		const cookieVal = authState[name];
+		// apply to request
+		request.state[name] = cookieVal.value;
+		// apply to response - note this special `reply` prop assigned onPreAuth
+		request.authorize.reply.state(name, cookieVal.value, cookieVal.opts);
```

Comment:
Clarification question -- Authentication is happening before the route is called, so regardless of whether that succeeds or fails, the router is still rendered and there's an attempt at making the associated queries still, yes? That part hasn't changed?